ignite/src/main/java/org/hibernate/ogm/datastore/ignite/IgniteDialect.java

```diff
@@ -117,19 +118,35 @@ else if ( lockMode == LockMode.OPTIMISTIC_FORCE_INCREMENT ) {
 	@Override
 	public Tuple getTuple(EntityKey key, OperationContext operationContext) {
 		IgniteCache<Object, BinaryObject> entityCache = provider.getEntityCache( key.getMetadata() );
-		if ( entityCache == null ) {
-			throw log.cacheNotFound( key.getMetadata().getTable() );
-		}
 		Object id = provider.createKeyObject( key );
-		BinaryObject po = entityCache.get( id );
-		if ( po != null ) {
-			return new Tuple( new IgniteTupleSnapshot( id, po, key.getMetadata() ), SnapshotType.UPDATE );
+		BinaryObject bo = entityCache.get( id );
+		if ( bo != null ) {
+			return new Tuple( new IgniteTupleSnapshot( id, bo, key.getMetadata() ), SnapshotType.UPDATE );
 		}
 		else {
 			return null;
 		}
 	}
 
+	@Override
+	public List<Tuple> getTuples(EntityKey[] keys, TupleContext tupleContext) {
+		List<Tuple> result = new ArrayList<>( keys.length );
+		IgniteCache<Object, BinaryObject> entityCache = provider.getEntityCache( keys[0].getMetadata() );
+		Map<EntityKey, Object> ids = new HashMap<>( keys.length );
```

Comment:
Maybe LinkedHashMap should be used here instead of HashMap? The order here seems to matter under PESSIMISTIC tx settings, where unordered getAll may easily cause deadlocks.

Reply:
I'm not sure but may be you right. Fixed it

Reply:
Thanks for the change! Also, here `idSet` will simply duplicate `ids` values as it will hold internally map and list, the same can be achieved using LinkedHashMap for `ids` and calling `entityCache.getAll( ids.values() )` thus requiring less intermediate structures allocation and method calls, as the LinkedHashMap is guaranteed to return values view that preserves insertion order.

Reply:
Your code isn't correct. getAll() requires Set, but values() is Collection.

Reply:
You are right. I forgot that. Just in case - `getAll ( Collection )` is available on the IgniteInternalCache which is returned by IgniteEx#cachex method and I see that IgnitionEx and IgniteEx are already used by IgniteDatastoreProvider.